packages/cli/src/util/progress.ts

```diff
@@ -0,0 +1,42 @@
+// zero is considered first index in the range
+type NeedleFunc = (needle: number) => void;
+
+export function showProgress({
+  total,
+  signal,
+  frequencyMs,
+  progress,
+}: {
+  total: number;
+  signal: AbortSignal;
+  frequencyMs: 1000;
+  progress: (opts: {current: number; total: number; percentage: number; ratePerSec: number}) => void;
+}): NeedleFunc {
+  let current = 0;
+  let last = 0;
+
+  const needle: NeedleFunc = (needle: number) => {
+    // zero is considered first index in the range
+    current = needle + 1;
```

Comment:
can we check against total here and finish immediately?

Reply:
We have to calculate the progress here as well but yes we can do that and finish immediately.

Reply:
@wemeetagain Just thought of it again, In case of finishing it right away we will not be able to calculate the `ratePerSec`, as we don't know in how much ms the needle was moved.

Reply:
I don't think we need the `ratePerSec` feature